launcher/ui/MainWindow.cpp

```diff
@@ -1137,6 +1145,11 @@ void MainWindow::showInstanceContextMenu(const QPoint &pos)
             connect(actionDeleteGroup, SIGNAL(triggered(bool)), SLOT(deleteGroup()));
             actions.append(actionDeleteGroup);
         }
+
+        QAction *actionUndoTrashInstance = new QAction("Undo last trash instance", this);
```

Comment:
This `actionUndoTrashInstance` is a new local variable for this function, and does NOT set the variable you declared here:

https://github.com/PolyMC/PolyMC/blob/8a7dec35ebf8f3f26d7ce92ef047b9b7d6e1754f/launcher/ui/MainWindow.cpp#L223

Therefore, `ui->actionUndoTrashInstance` is null, and when you do anything on it, it crashes.

There are at least two ways to fix this:

1. Store one `actionUndoTrashInstance` in the `MainWindow`, only create it once (otherwise the action's enabled state will be reset every time you remake it), and change its enable state when something is trashed as you have already done.
2. Remove the MainWindow's `actionUndoTrashInstance`, and add code right here to set the action as enabled/disabled as necessary. (Remove the other `ui->actionUndoTrashInstance->setEnabled();` calls if you choose to do this.)

Reply:
how do you do that? I'm thinking of making `InstanceList` emit an event when the `trashStack` is updated, and then if `APPLICATION->instanceList()->trashStack.isEmpty()` disable, else enable

but then would it require a separate slot function to `connect` to?

Reply:
From what I understand, `actionUndoTrashInstance` is recreated every time you right-click and show the menu. So there's really nothing special you need to do (unless I misunderstand how this works) – try using `setEnabled` right here based on whether the trash stack is empty, and see if it works as intended.

Reply:
I would prefer if this were an option in the menubar as well